target/linux/ipq806x/base-files/etc/hotplug.d/firmware/11-ath10k-caldata

```diff
@@ -9,6 +9,10 @@ board=$(board_name)
 case "$FIRMWARE" in
 "ath10k/pre-cal-pci-0000:01:00.0.bin")
 	case $board in
+	asrock,g10)
+		ath10kcal_extract "0:ART" 0x1000 0x2f20
+		ath10k_patch_mac $(mtd_get_mac_binary 0:ART 0x1006)
```

Comment:
If caldata is 0x1000, the address from 0x1006 should be read automatically and you can drop the ath10k_patch_mac. Can you check that?
Same below for 0x5006 ...

Reply:
Without it MAC addresses are random:

````
br-lan    Link encap:Ethernet  HWaddr 74:DA:38:6A:65:F5  
eth0      Link encap:Ethernet  HWaddr 74:DA:38:6A:65:F4  
eth0.2    Link encap:Ethernet  HWaddr 74:DA:38:6A:65:F4  
eth1      Link encap:Ethernet  HWaddr 74:DA:38:6A:65:F5  
eth1.1    Link encap:Ethernet  HWaddr 74:DA:38:6A:65:F5  
wlan0     Link encap:Ethernet  HWaddr 00:34:56:78:EB:EB  
wlan1     Link encap:Ethernet  HWaddr 00:34:56:78:5B:5B
````

````
root@OpenWrt:/# ifconfig  -a | grep  HW
br-lan    Link encap:Ethernet  HWaddr 74:DA:38:6A:65:F5  
eth0      Link encap:Ethernet  HWaddr 74:DA:38:6A:65:F4  
eth0.2    Link encap:Ethernet  HWaddr 74:DA:38:6A:65:F4  
eth1      Link encap:Ethernet  HWaddr 74:DA:38:6A:65:F5  
eth1.1    Link encap:Ethernet  HWaddr 74:DA:38:6A:65:F5  
wlan0     Link encap:Ethernet  HWaddr 00:34:56:78:14:14  
wlan1     Link encap:Ethernet  HWaddr 00:34:56:78:6D:6D
````

Reply:
Okay, but I don't get it. From how I understand it, ath10k_patch_mac will overwrite the MAC address with itself.

Reply:
I'm not ath10k specialist, but every device in OpenWrt with ath10k have "ath10k_patch_mac" line in 11-ath10k-caldata file.

Reply:
No, e.g. https://github.com/openwrt/openwrt/blob/master/target/linux/ath79/generic/base-files/etc/hotplug.d/firmware/11-ath10k-caldata

The ath10k driver will take the MAC address from an offset 6 of the firmware file. For some devices, no usuable MAC address or just a dummy MAC address is left there, so we need to take the address from somewhere _and write it to that position with offset 6 in the firmware file_.

That's why I don't understand why that line is necessary in your case: You read the firmware file from flash at 0x1000 and then copy it somewhere else. Then you take the MAC address from 0x1006 and right it to that firmware file with an offset of 6, i.e. where it should already be, as it's been copied from the very same source.

So, I seem to be missing a piece.

Reply:
I'm not sure if it's relevant, but you should use `caldata_extract` here anyway.

Reply:
@mans0n Hero of the day. There is no ath10kcal_extract function anymore, so this just did nothing, and then the second line just put the MAC address in an empty firmware file!

Reply:
https://github.com/openwrt/openwrt/commit/5b6a80909280cafcb1e28ca120eed6922d68dc5a

Reply:
@CHKDSK88 So, you will just need
`caldata_extract "0:ART" 0x1000 0x2f20`
here without *patch_mac and the equivalent for the second interface.

Reply:
@mans0n 

Thank You. You rescued me again. :)